addons/popochiu/engine/interfaces/i_room.gd

```diff
@@ -110,13 +117,197 @@ func clear_instances() -> void:
+		var chr: PopochiuCharacter = C.get_character(chr_script_name)
+		
+		if not chr: continue
+		
+		chr.position = Vector2(chr_dic.x, chr_dic.y)
+		chr._looking_dir = chr_dic.facing
+		chr.visible = chr_dic.visible
+		chr.modulate = Color.from_string(chr_dic.modulate, Color.WHITE)
+		chr.self_modulate = Color.from_string(chr_dic.self_modulate, Color.WHITE)
+		chr.light_mask = chr_dic.light_mask
+		
+		current.add_character(chr)
+	
+	# If the room must have the player character but it is not part of its $Characters node, then
+	# add the PopochiuCharacter to the room
+	if (
+		current.has_player
+		and is_instance_valid(C.player)
+		and not current.has_character(C.player.script_name)
+	):
+		current.add_character(C.player)
+		# Place the PC in the middle of the room
+		C.player.position = Vector2(E.width, E.height) / 2.0
+		await C.player.idle()
+	
+	# Load the state of Props, Hotspots, Regions and WalkableAreas
+	for type in PopochiuResources.ROOM_CHILDS:
+		for script_name in rooms_states[room.script_name][type]:
+			var node: Node2D = current.callv(
+				"get_" + type.trim_suffix("s"),
+				[(script_name as String).to_pascal_case()]
+			)
+			var node_dic: Dictionary =\
+			rooms_states[room.script_name][type][script_name]
+			
+			for property in node_dic:
+				if not PopochiuResources.has_property(node, property): continue
+				
+				node[property] = node_dic[property]
+	
+	for c in get_tree().get_nodes_in_group("PopochiuClickable"):
+		c.room = current
+	
+	await current._on_room_entered()
+	
+	if E.loaded_game:
+		C.player.global_position = Vector2(
+			E.loaded_game.player.position.x,
+			E.loaded_game.player.position.y
+		)
+	
+	if _use_transition_on_room_change:
+		E.tl.play_transition(E.tl.FADE_OUT)
+		await E.tl.transition_finished
+		
+		await E.wait(0.3)
+	else:
+		await get_tree().process_frame
+	
+	if not current.hide_gi:
+		G.unblock()
+	
+	if E.hovered:
+		G.mouse_entered_clickable.emit(E.hovered)
+	
+	E.in_room = true
+	
+	if E.loaded_game:
+		E.game_loaded.emit(E.loaded_game)
+		await G.show_system_text("Game loaded")
+		
+		E.loaded_game = {}
+	
+	# This enables the room to listen input events
+	current.is_current = true
+	await current._on_room_transition_finished()
+	
+	# Fix #219: Update visited_first_time state once _on_room_transition_finished() finishes
+	current.state.visited_first_time = false
+
+
+func store_states() -> void:
+	# Store the default state of rooms in the game
+	for room_tres in PopochiuResources.get_section("rooms"):
+		var res: PopochiuRoomData = load(room_tres)
+		rooms_states[res.script_name] = res
+		res.save_childs_states()
+
+
 #endregion
 
 #region SetGet #####################################################################################
 func set_current(value: PopochiuRoom) -> void:
-	current = value
-	
-	E.goto_room(current.script_name)
+	if not value.is_inside_tree():
+		goto_room(value.script_name)
+		#value.queue_free()
```

Comment:
if this is not invoked I would clean this up

Reply:
Done.